packages/remix-dev/vite/plugin.ts

```diff
@@ -796,10 +809,27 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
         });
         await viteChildCompiler.pluginContainer.buildStart({});
       },
-      transform(code, id) {
+      async transform(code, id) {
```

Comment:
Reading through the [Vite conventions for virtual modules](https://vitejs.dev/guide/api-plugin#virtual-modules-convention) again and wondering if we should do something to ensure the client routes don't get processed/handled by other plugins. We could just prepend with `\0` or model them fully as vmods.

Reply:
I think it should be okay if it's handled by other plugins. In terms of resolving the module, since it's a real file on disk, it'll be treated the same as the original route file. In terms of transforming the file, it's going to behave as if you'd manually written a re-export file, which also would work fine.

The benefit of this approach from what I understand is that Vite will treat this pseudo-virtual module the same as the route file in terms of file changes, invalidation, HMR. This means we don't need to do anything special to invalidate virtual modules when the underlying route changes.